src/python/pants/reporting/streaming_workunit_handler.py

```diff
@@ -48,7 +48,7 @@ def session(self) -> Iterator[None]:
 
 class _InnerHandler(threading.Thread):
   def __init__(self, scheduler: Any, callbacks: Iterable[Callable], report_interval: float):
-    super(_InnerHandler, self).__init__()
+    super(_InnerHandler, self).__init__(daemon=True)
```

Comment:
Nit: this should be `super().__init__` for less verbosity. It only isn't because the script we used to update our `super()` calls when we dropped Py2 support didn't account for a leading `_` in the regex.